lib/src/reducers/strands_reducer.dart

```diff
@@ -223,7 +223,8 @@ Strand move_strand(
     Substrand substrand = substrands[i];
     Substrand new_substrand = substrand;
     if (substrand is Domain) {
-      Domain mirrored_domain = domains[domains.length - counter - 1];
+      Domain mirrored_domain = domains[counter];
```

Comment:
For the record, I think the bug was introduced in commit https://github.com/UC-Davis-molecular-computing/scadnano/commit/1142361dd4354a6751df8740f0f85b21c523db67 when fixing issue #467.

The issue there was that if we reversed the orientation of the strand, e.g., the top helix was forward and now it's reversed, then the domain names should have reversed along the whole strand. But the fix was too broad: it simply reversed the order of domain names unconditionally when moving strands, whether reversing the orientation or not.

In some sense it's not entirely clear what should happen when the orientation is reversed, actually. For instance, consider this design:

![image](https://user-images.githubusercontent.com/19274365/141321586-1800b8bf-c5d1-4231-a967-1c36ad009576.png)

Reverse the orientation:

![image](https://user-images.githubusercontent.com/19274365/141321616-8edc3d5e-e132-45df-95e9-4e46da39f9fd.png)

Note that XYZ had length 4 originally, and now it has length 8. Is that the intended behavior or what the user should expect? It's not clear. Even in my original example in #467:

![image](https://user-images.githubusercontent.com/19274365/141321902-4e057c6d-5c1e-404d-9ec6-4d0cca246d05.png)

![image](https://user-images.githubusercontent.com/19274365/141321918-fd3c9fb1-2462-4b74-914d-0e8fd04c0ade.png)

I thought it was "obvious" that the domain names should appear in the opposite order (i.e., that the second image above was a bug), because I was thinking that the user's intent is for the names to be given in order 5' to 3'.

But even that example, each domain name has its length appended to the end, and after switching the orientation above, the lengths still match the domain names: for instance, the length 11 domain ends in `_11` both before and after. With the fix for issue #467, now that's no longer true.

Anyway, I think the behavior in this PR is perfectly fine. It's not clear there's one "correct" thing to do when orientation is reversed. But if it is not reversed, then the behavior in this PR is definitely the correct one.